parser-typechecker/src/Unison/DataDeclaration.hs

```diff
@@ -487,6 +522,26 @@ pattern TupleType' ts <- (unTupleType -> Just ts)
 pattern TupleTerm' xs <- (unTupleTerm -> Just xs)
 pattern TuplePattern ps <- (unTuplePattern -> Just ps)
 
+-- some pattern synonyms to make pattern matching on some of these constants more pleasant 
+pattern DocRef <- ((== docRef) -> True) 
+pattern DocJoin segs <- Term.App' (Term.Constructor' DocRef DocJoinId) (Term.Sequence' segs)
+pattern DocBlob txt <- Term.App' (Term.Constructor' DocRef DocBlobId) (Term.Text' txt)
+pattern DocLink link <- Term.App' (Term.Constructor' DocRef DocLinkId) link
+pattern DocSource link <- Term.App' (Term.Constructor' DocRef DocSourceId) link
+pattern DocSignature link <- Term.App' (Term.Constructor' DocRef DocSignatureId) link
+pattern DocEvaluate link <- Term.App' (Term.Constructor' DocRef DocEvaluateId) link
+pattern DocSignatureId <- ((== docSignatureId) -> True)
+pattern DocBlobId <- ((== docBlobId) -> True)
+pattern DocLinkId <- ((== docLinkId) -> True)
+pattern DocSourceId <- ((== docSourceId) -> True)
+pattern DocEvaluateId <- ((== docEvaluateId) -> True)
+pattern DocJoinId <- ((== docJoinId) -> True)
+pattern LinkTermId <- ((== linkTermId) -> True)
+pattern LinkTypeId <- ((== linkTypeId) -> True)
+pattern LinkRef <- ((== linkRef) -> True)
+pattern LinkTerm tm <- Term.App' (Term.Constructor' LinkRef LinkTermId) tm 
+pattern LinkType ty <- Term.App' (Term.Constructor' LinkRef LinkTypeId) ty 
```

Comment:
Here's where those pattern synonyms are introduced.

Reply:
Like this. 😄